actions/push-to-dockerhub/action.yaml

```diff
@@ -0,0 +1,63 @@
+name: Publish Docker image
+description: Publish docker images to DockerHub
+
+inputs:
+  repository:
+    description: |
+      The caller workflow's repository
+  tags:
+    description: |
+      List of Docker images to be pushed.
+    required: true
+  build_path:
+    description: |
+      Path to Dockerfile, which is used to build the image.
+    default: "."
+
+runs:
+  using: composite
+  steps:
+    - name: Check out the repo
+      uses: actions/checkout@v4
+
+    - name: Get secrets for DockerHub login
+      id: get-secrets
+      uses: grafana/shared-workflows/actions/get-vault-secrets@main
+      with:
+        common_secrets: |
+          DOCKERHUB_USERNAME=dockerhub:username
+          DOCKERHUB_PASSWORD=dockerhub:password
+
+    - name: Log in to Docker Hub
+      uses: docker/login-action@f4ef78c080cd8ba55a85445d5b36e214a81df20a
```

Comment:
Would it be possible to limit the login step somehow to only specific CI events? Like that it's only done when the image would also be pushed?

Reply:
I think this should be specified on the caller, no?

Reply:
True, but isn't there something similar down below when setting the `push` property of the "Build and push Docker image" step?

Reply:
Hm, I think if we agree that DockerHub will be only used for "prod" images then we can simply fail early if the event is not a push. Then all the "dev" images and images used for tooling etc should live in GAR. But I guess it's going to be some time before this happens 😅 Happy if we leave it as is and create an issue to address that ASAP?

Reply:
Absolutely 🙂

Reply:
https://github.com/grafana/shared-workflows/issues/27 @zerok fyi :)